device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/InternalClient.java

```diff
@@ -522,21 +524,22 @@ void startTwinInternal(IotHubEventCallback twinStatusCallback, Object twinStatus
             throw new IOException("Open the client connection before using it.");
         }
 
-        if (twinStatusCallback == null || genericPropertyCallBack == null)
+        if (this.twin != null && twinStatusCallback == null && genericPropertyCallBack == null)
         {
-            //Codes_SRS_INTERNALCLIENT_34_082: [If either callback is null, this function shall throw an IllegalArgumentException.]
-            throw new IllegalArgumentException("Callback cannot be null");
+            //end twin
+            twin.tearDownTwin();
         }
-        if (this.twin == null)
+        else if (this.twin == null)
```

Comment:
if (this.twin == null) && (twinStatusCallback == null), we should do nothing?

Reply:
We should probably throw an illegalstateexception or something instead. If the call to a method called "startTwin" returns without error, customers may get confused. It should only return with success if it can build up twin or tear down twin